cylc/flow/task_pool.py

```diff
@@ -960,90 +1022,171 @@ def release_all_tasks(self):
+        # (see also the similar check + log message in scheduler.py)
+        if self.config.initial_point and point < self.config.initial_point:
+            # Attempted manual trigger prior to FCP
+            # or future triggers like foo[+P1] => bar, with foo at ICP.
+            LOG.debug(
+                'Not spawning %s.%s: before initial cycle point', name, point)
+            return False
+        elif self.config.final_point and point > self.config.final_point:
+            # Only happens on manual trigger beyond FCP
+            LOG.debug(
+                'Not spawning %s.%s: beyond final cycle point', name, point)
+            return False
 
-        Return the number of removed tasks.
-        """
+        return True
+
+    def spawn_task(self, name, point, flow_label=None, reflow=True,
+                   parent_id=None):
+        """Spawn name.point and add to runahead pool. Return it, or None."""
+        if not self.can_spawn(name, point):
+            return None
+
+        # Get submit number by flow label {flow_label: submit_num, ...}
+        snums = self.suite_db_mgr.pri_dao.select_submit_nums(name, str(point))
+        try:
+            submit_num = max(snums.values())
+        except ValueError:
+            # Task never spawned in any flow.
+            submit_num = 0
+
+        for f_id in snums.keys():
+            # Flow labels of previous instances.  E.g. f_id "u".
+            if self.flow_label_mgr.match_labels(flow_label, f_id):
+                # Already spawned in this flow. E.g. flow_label "uV".
+                # TODO update existing DB row to avoid cond reflow from V too?
+                LOG.warning('Not spawning %s.%s (spawned in flow %s)',
+                            name, point, f_id)
+                return None
+
+        # Spawn if on-sequence and within recurrence bounds.
+        taskdef = self.config.get_taskdef(name)
+        if not taskdef.is_valid_point(point):
+            return None
+
+        itask = TaskProxy(
+            taskdef,
+            point, flow_label,
+            submit_num=submit_num, reflow=reflow)
+        if parent_id is not None:
+            msg = "(" + parent_id + ") spawned %s.%s flow(%s)"
+        else:
+            msg = "(no parent) spawned %s.%s %s"
+        if flow_label is None:
+            # Manual trigger: new flow
+            msg += " (new flow)"
+
+        # Hold task if beyond the suite hold point
+        if self.hold_point and itask.point > self.hold_point:
+            LOG.info(
+                "[%s] -holding (beyond suite hold point) %s",
+                itask, self.hold_point)
+            itask.state.reset(is_held=True)
+        elif (self.stop_point and itask.point <= self.stop_point and
+                self.task_has_future_trigger_overrun(itask)):
+            LOG.info("[%s] -holding (future trigger beyond stop point)", itask)
+            self.held_future_tasks.append(itask.identity)
+            itask.state.reset(is_held=True)
```

Comment:
Good question, I can't recall why we do this - but note this is also done on master (this code block just got relocated on this branch). I'll just test what happens if I don't hold such a task...

Reply:
Ah, it appears to be a deliberate thing to allow us to avoid stalling a suite with a task waiting on future trigger beyond the FCP. Example case:
```
[scheduling]
    cycling mode = integer
    initial cycle point = 1
    final cycle point = 3
    [[graph]]
        P1 = """
           foo & bar
           foo[+P1] & bar => baz
           """
[runtime]
    [[root]]
        script = sleep 2
```
With the code as above, this suite shuts down normally with a held waiting `baz.3` in the task pool.  

By if I comment out the reset-to-waiting it stalls with: 
```
2020-07-22T17:40:15+12:00 WARNING - suite stalled
2020-07-22T17:40:15+12:00 WARNING - Unmet prerequisites for baz.3:
2020-07-22T17:40:15+12:00 WARNING -  * foo.4 succeeded
```
You could think of this as a cheap equivalent of our pre-initial-dependency-ignore system at the other end of the graph.

As it happens, especially in SoD, waiting tasks aren't that useful as a "stall" indicator anyway so we may change this in the future. For now I'll just add a comment to the code...

Reply:
But aren't you still missing the `itask.point > self.stop_point` case, where your spawned task is beyond the stop point? Shouldn't this also be held? 

Also `self.task_has_future_trigger_overrun` also checks for non-null `self.stop_point`.. So, my suggested could also be:
```
        elif self.stop_point and itask.point > self.stop_point:
            LOG.info("[%s] -holding (task spawned beyond stop point)", itask)
            self.held_future_tasks.append(itask.identity)
            itask.state.reset(is_held=True)
        elif self.task_has_future_trigger_overrun(itask):
            LOG.info("[%s] -holding (future trigger beyond stop point)", itask)
            self.held_future_tasks.append(itask.identity)
            itask.state.reset(is_held=True)
```
This also avoids stalling with your example  (maybe I'm missing something?)

Reply:
> But aren't you still missing the itask.point > self.stop_point case, where your spawned task is beyond the stop point? Shouldn't this also be held?

No, there's no need to hold tasks beyond the stop point because those do not get released from the runahead pool.

Reply:
If the stop point is the FCP it makes no difference whether or not we hold these tasks, because they can never run anyway (they depend on tasks beyond that FCP that will never exist).

However, it does matter if the stop point is less than the FCP, because these tasks will then be held if we restart the suite to carry on beyond the initial stop point.  I can't actually think of any good reason to hold them. The auto shutdown check looks for them in the `held_future_tasks` list but it does not care if they're held or not.

So it's not the problem of this branch, but an easy change so I'll do it: I'll rename `held_future_tasks` to `stuck_future_tasks` and quit holding them.

Reply:
Hmmm, actually I need to think this through: the `held` state is to avoid identifying these tasks as the cause of a stall; but the `held_future_tasks` list is avoid identifying them as a reason not to shut down, which are slightly different purposes.  I might punt this to a follow-up issue since it is not SoD-specific.

Reply:
Ok, will leave this one to you I suppose (^.^)... The fact they stay runahead is good enough to me, if only those specific tasks before/at the stop point could be in the runahead also! (but I guess they are cycle specific).

Reply:
Fix in latest commits.